_episodes/01-introduction.md

```diff
@@ -79,16 +81,23 @@ instruction-intensive operations in order to improve the overall performance
 compared with both sequential and homogeneous parallel programming models.
 
 To better understand the concept of heterogeneous parallel programming,
-we should introduce the following concepts:
+we will introduce the following concepts from computer networking:
 
-- **Latency**: The duration of an operation from its beginning to its completion expressed in microseconds (*m*s)
-- **Throughput**: The number of operations processed per unit of time expressed in gigaflops (*G*flops), which
-translates into a billion floating-point operations per second.
-- **Bandwidth**: The processed amount of data per unit of time expressed in megabytes per second (*M*B/s) 
-or gigabytes per second (*G*B/s)
+- **Latency**: The duration of an operation from its beginning to its completion expressed in units of time e.g. microseconds (*m*s)
+- **Bandwidth**: The *theoretical* capacity of how much data could be transferred from source to destination per unit of time,
+  commonly expressed in gigabytes per second (*G*B/s) for GPUs  
+- **Throughput**: The *real* measure of how much data is successfully transferred from source to destination per unit of time, 
+expressed in gigaflops (*G*flops) for GPUs, which translates into a billion floating-point operations per second. Therefore, throughput 
+is always less than the bandwidth.
+
+![Figure 0](../fig/pipe.png)
+
+One way of qualitatively understanding the 3 concepts is by visualizing water flowing through a pipe. The water represents "data", and 
+the pipe diameter represents the "bandwidth", while the amount of water flowing through the pipe over a period of time
+represents the "throughput".
```

Comment:
I like these changes and I support their merge. However, I would remove any generalization statements such as "_Therefore, throughput is always less than the bandwidth._" no matter how practical they could be. That's is not a good place for it within the definitions anyway.

Reply:
It's a fact that throughput < bandwidth, no assumptions made here. However, given that you're measuring the throughput in Gflops, I think you're prolly referring to the computational  throughput which is different altogether from the bandwidth [that is often used by CUDA users]. Whichever terminology you use, I suggest you tailor this to a simple code that uses bandwidth/throughput to differentiate between compute and memory bound kernels [the latter being very common in practice].